### PR TITLE
provider: make existing provider managers can be reused

### PR DESCRIPTION
**Pull request by liudalibj**
- Add new GetConfig function

Fixes https://github.com/confidential-containers/cloud-api-adaptor/issues/1628

### CODE REVIEW

**Comment by liudalibj** [comment]
`https://libvirt.org/` can be accessed from my dev machine, didn't find out what's make https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/7270996164/job/19810981198?pr=1629 FAIL.

**Comment by liudalibj** [comment]
> `https://libvirt.org/` can be accessed from my dev machine, didn't find out what's make https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/7270996164/job/19810981198?pr=1629 FAIL.

Re-run the failed task, it's PASS now.

File: libvirt/install_operator.sh [hunk 1]
```diff
@@ -11,11 +11,11 @@ set -o pipefail
 
 script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
-LIBVIRT_IP="${LIBVIRT_IP:-}"
+LIBVIRT_IP="${LIBVIRT_IP:-192.168.122.1}"
```
Comment (mkulke) [inline libvirt/install_operator.sh:14]: those seem unrelated. can we move them to their own PR?

Comment (liudalibj) [reply libvirt/install_operator.sh]: remove unrelated things, will put them ti a new PR.

**Review by mkulke — APPROVED** [review]

**Review by huoqifeng — APPROVED** [review]
LGTM

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks!
